mltsp/predict.py

```diff
@@ -1,287 +1,42 @@
-                    fname = els[0]
-                    meta_feats = els[1:]
-                    for i in range(len(meta_feats)):
-                        try:
-                            meta_feats[i] = float(meta_feats[i])
-                        except ValueError:
-                            pass
-                    meta_features[fname] = dict(
-                        list(zip(meta_feat_names, meta_feats)))
-    else:
-        meta_features = {}
-    return meta_features
-
-
-def determine_feats_used(featset_key):
-    """
-    """
-    with open(os.path.join(
-            cfg.FEATURES_FOLDER, "%s_features.csv" % featset_key)) as f:
-        features_in_model = f.readline().strip().split(',')
-    return features_in_model
-
-
-def featurize_multiple_serially(newpred_file_path, tmp_dir_path,
-                                features_to_use, custom_features_script,
-                                all_meta_features, sep=","):
-    """
-    """
-    the_tarfile = tarfile.open(newpred_file_path)
-    all_fnames = the_tarfile.getnames()
-    the_tarfile.extractall(path=tmp_dir_path)
-    big_features_and_tsdata_dict = {}
-    for fname in all_fnames:
-        meta_features = all_meta_features.get(os.path.basename(fname), {})
-        big_features_and_tsdata_dict.update(featurize_single(
-            fname, features_to_use, custom_features_script, meta_features,
-            tmp_dir_path=tmp_dir_path, sep=sep))
-    return big_features_and_tsdata_dict
-
-
-def featurize_single(newpred_file_path, features_to_use, custom_features_script,
-                     meta_features, tmp_dir_path="/tmp", sep=","):
-    """
-    """
-    fname = newpred_file_path
-    if os.path.isfile(fname) and os.path.isabs(fname):
-        filepath = fname
-    elif os.path.isfile(os.path.join(tmp_dir_path, fname)) and \
-         os.path.isabs(os.path.join(tmp_dir_path, fname)):
-        filepath = os.path.join(tmp_dir_path, fname)
-    elif os.path.isfile(os.path.join(os.path.join(cfg.UPLOAD_FOLDER,
-                                                  "unzipped"),
-                                     fname)) and \
-         os.path.isabs(os.path.join(os.path.join(cfg.UPLOAD_FOLDER, "unzipped"),
-                                    fname)):
-        filepath = os.path.join(os.path.join(cfg.UPLOAD_FOLDER, "unzipped"),
-                                fname)
-    elif os.path.isfile(os.path.join(cfg.UPLOAD_FOLDER, fname)) and \
-         os.path.isabs(os.path.join(cfg.UPLOAD_FOLDER, fname)):
-        filepath = os.path.join(cfg.UPLOAD_FOLDER, fname)
+from . import featurize
+from . import featurize_tools as ft
+from . import util
+
+
+def model_predictions(featureset, model):
+    """Construct a pandas.DataFrame of model predictions for a given featureset."""
+    # Do probabilistic model prediction when possible
+    feature_df = build_model.rectangularize_featureset(featureset)
+    if issubclass(type(model), ClassifierMixin):
+        try:
```

Comment:
This logic is a bit weird.  Why not just try `predict_proba` and then `predict`, regardless of the type of the model?